File: pgm/ABC_Java/ABC_245_A.java
import java.io.*;
// import java.math.*;
import java.util.*;
// import java.util.Map.*;
// import java.util.stream.*;

public class Main {
    public static void main (String[] args) throws Exception {
        BufferedReader reader =
            new BufferedReader (new InputStreamReader (System.in));
        final int[] arr = Arrays.stream (reader.readLine().split (" "))
                              .mapToInt (Integer::parseInt)
                              .toArray();
        final String Takahashi = "Takahashi";
        final String Aoki = "Aoki";
        final int A = arr[0];
        final int B = arr[1];
        final int C = arr[2];
        final int D = arr[3];
        if (A < C) {
            System.out.println (Takahashi);
        } else if (A > C) {
            System.out.println (Aoki);
        } else if (B <= D) {
            System.out.println (Takahashi);
        } else {
            System.out.println (Aoki);
        }
        return;
    }
}

File: pgm/ABC_Kotlin/ABC_245_A.kt
// import java.io.BufferedOutputStream
// import java.io.BufferedReader
// import java.io.DataInputStream
// import java.io.FileInputStream
// import java.io.IOException
// import java.io.InputStreamReader
// import java.io.PrintWriter
// import java.lang.StringBuilder
// import java.time.LocalTime
// import java.time.format.DateTimeFormatter
// import java.util.Scanner
// import java.util.StringTokenizer

// import kotlin.math.*

const val Takahashi: String = "Takahashi"
const val Aoki: String = "Aoki"
const val space: String = " "

fun main() {
    val (A: Int, B: Int, C: Int, D: Int) = readln().split(space).map { it.toInt() }
    if (A < C) {
        println(Takahashi)
    } else if (A > C) {
        println(Aoki)
    } else if (B <= D) {
        println(Takahashi)
    } else {
        println(Aoki)
    }
    kotlin.system.exitProcess(0)
}